tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -590,30 +603,83 @@ void TangoRosNode::PublishStaticTransforms() {
 }
 
 void TangoRosNode::OnPoseAvailable(const TangoPoseData* pose) {
-  if (pose->frame.base == TANGO_COORDINATE_FRAME_START_OF_SERVICE
-      && pose->frame.target == TANGO_COORDINATE_FRAME_DEVICE) {
+  if (framePairMatches(pose->frame, TANGO_COORDINATE_FRAME_START_OF_SERVICE,
+                       TANGO_COORDINATE_FRAME_DEVICE)) {
+    if (pose->status_code == TANGO_POSE_VALID && pose_available_mutex_.try_lock()) {
+      if (localization_mode_ == (int) LocalizationMode::ODOMETRY) {
+        // Localization is not used, publish device wrt. start_of_service.
+        toTransformStamped(*pose, time_offset_, &start_of_service_T_device_);
+        start_of_service_T_device_.header.frame_id =
+            toFrameId(TANGO_COORDINATE_FRAME_START_OF_SERVICE);
+        start_of_service_T_device_.child_frame_id =
+            toFrameId(TANGO_COORDINATE_FRAME_DEVICE);
+      } else if (localization_status_ == LocalizationStatus::LOCALIZING) {
+        // We are not localized yet.
+        LOG(INFO) << "Not localized yet, please walk around.";
+      } else if (localization_status_ == LocalizationStatus::LOCALIZATION_LOST) {
+        LOG(WARNING) << "Recovering tracking...";
```

Comment:
It can happen if the adf_T_device pose is invalid and we still did not get any valid sos_T_adf pose. Is it something that could actually happen in Tango?